compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/fuse_fill_into_forall.mlir

```diff
@@ -29,3 +29,43 @@ func.func @fuse_fill_into_forall(%arg0: tensor<1x4x16x64xi8>, %arg1 : tensor<4x1
 // CHECK:     linalg.fill
 // CHECK:     linalg.generic
 // CHECK:  }
+
+// -----
+
+#map = affine_map<(d0) -> (d0)>
+
+// CHECK: @fuse_without_slice(%[[FUNCARG:.*]]: tensor<8xi8>) -> tensor<8xi8> {
+func.func @fuse_without_slice(%arg0: tensor<8xi8>) -> tensor<8xi8> {
+  %c7_i8 = arith.constant 7 : i8
+  %c3_i8 = arith.constant 3 : i8
+  %0 = linalg.fill ins(%c7_i8 : i8) outs(%arg0 : tensor<8xi8>) -> tensor<8xi8>
+  %1 = tensor.empty() : tensor<8xi8>
+
+  // check that the operand of scf.forall is not the filled tensor, because the
+  // fill will take place inside the scf.forall.
+  // CHECK: %[[FORALL:.*]] = scf.forall (%[[ARG1:.*]]) in (1)
+  // CHECK-SAME: shared_outs(%[[ARG2:.*]] = %[[FUNCARG]])
+
+  // check for the new fill
+  // CHECK: %[[NEWFILL:.*]] = linalg.fill
+  // CHECK-SAME: outs(%[[ARG2]] : tensor<8xi8>) -> tensor<8xi8>
```

Comment:
I feel it's not readable if you mixed `CHECK` with the comments. I'd prefer you put all the comments above and keep `//CHECK` section compact.

Reply:
Updated